### ISSUE TITLE
ZeroClipboard problem with complex style

### ISSUE THREAD

**Issue by rothanachoun** [zeroclipboard/zeroclipboard#536]
I have a drop down menu icon that show when I move mouse over on a card. In drop down menu I have a 'Copy Link' (clip element) and when I over on it, it hide my drop down menu icon include 'Copy Link' because the transparent flash. 
Below is my layout
![clippy2](https://cloud.githubusercontent.com/assets/5331630/5649372/d1ba1038-96c7-11e4-81f1-a3793e22e75a.png)

**Comment by JamesMGreene** [comment]
To ensure I am understanding your setup:

 1. Are the following repro steps correct?
     1. You mouseover/hover over a "card" item, the dropdown opener icon is shown [but not the dropdown contents]
     2. You click the dropdown opener icon, the dropdown contents are shown
     3. You mouseover/hover over the "Copy Link" dropdown item (which is then overlaid by ZeroClipboard's Flash object), then both the dropdown contents and the dropdown opener icon are immediately hidden

 2. Is that dropdown something custom that you built or from a more common open source library (e.g. Bootstrap, jQuery UI, etc.)?  The implementation details of its show/hide functionality are extremely critical to making it work in conjunction with ZeroClipboard but usually achievable.

**Comment by JamesMGreene** [comment]
3. Any chance you can create a simplified test page and/or JSBin that reproduces this problem? That always help speed us along in assisting you.

**Comment by rothanachoun** [comment]
@JamesMGreene Yeah, the following repro steps are correct. 
I use Semantic-UI dropdown menu.

**Comment by rothanachoun** [comment]
Here is the problem that I reproduced on JSFiddle.
http://jsfiddle.net/koa04mju/12/

Great thanks for your fast response.

**Comment by JamesMGreene** [comment]
Right, so... this is a very unfortunate scenario that is hard to avoid because CSS selectors are still extremely limited when it comes to recognizing child relationships.  For example, CSS4 (next version) is currently slated to support a `:has` pseudo-selector would you would, in theory, be able to use for this exact scenario, e.g.:

```css
.card:hover, .card:has(> .clippy.zeroclipboard-is-hover) {
    background: #ddd;
}

.card:hover .clippy, .clippy.zeroclipboard-is-hover {
    display: block;
}
```

However, even with that `:has` selector being specified, the above will **NOT** actually be possible because the `:has` selector is slated to only be added to the JavaScript API for CSS (a.k.a. "slow/full/complete profile", "CSS Object Model (CSSOM)") and NOT supported in normal CSS rendering (a.k.a. "fast profile") due to performance concerns.  How disappointing is that? :confounded:

So, long story short, this leaves us with a few distinct possibilities for this particular issue to be addressed:

 1. You change your CSS/UI to be driven by JavaScript `MouseEvent` handlers rather than CSS `:hover` selectors. This is a lame option for you but, honestly, might be the most maintainable option in the end.

 2. You leave your CSS/UI as-is but also setup a few JavaScript `MouseEvent` handlers to provide 
additional "overriding" inline CSS styles temporarily as needed. For example:
     ```js
var $clippy = $("#copyMe");
new ZeroClipboard($clippy);

$clippy
  .on("mouseenter", function(e) {
    // Temporarily force an inline CSS `display` style for the parent card/dropdown, which will
    // override the external CSS stylesheet's `display: none` rule for the non-`:hover` state.
    e.currentTarget.parentNode.style.display = "block";
  })
  .on("mouseleave", function(e) {
    // Remove the inline CSS `display` style for the parent card/dropdown, which will allow the
    // external CSS stylesheet's `display` rules to apply again
    e.currentTarget.parentNode.style.display = "";
  });
```
     Unfortunately, for this to consistently work (cross-browser, at least), we (ZeroClipboard) will first need to finish my in-progress branch/PR for fixing #463 (normalizing and de-duplicating `MouseEvent` dispatching for clipped elements while the Flash object is activated on top of them). While work on that is looking promising, it is still currently proving to be frustratingly difficult to get working consistently cross-browser. If you have time to help, I'm sure a second set of eyes would probably be beneficial.

 3. You use "ZeroClipboard.Core.js" instead of "ZeroClipboard.js" and deal with positioning the Flash object element yourself (i.e. you could choose to append it to the dropdown menu's DOM such that, even when actively placed over your clipped element, it would still be a descendent of the card and thus the `:hover` styles would remain active. There are more cons than pros to placing the Flash object element as described (moving it into the relative DOM) when choosing the positioning tactic for ZeroClipboard as a generically usable library ("ZeroClipboard.js")... but, for specific HTML and CSS setups (like your own) that only need to work in more modern browsers (hopefully like your page :sweat_smile:), then it can work very effectively with minimal negative impacts (the main one to watch out for [in modern browsers] are incorrect style rendering of `:nth-child`, `:first-child`, `:last-child` CSS rules due to the Flash object element/wrapper being inserted into the relative DOM hierarchy). We publish a separate "ZeroClipboard.Core.js" pretty much specifically for custom DOM needs like this, and so 3rd party wrappers can be produced with customized APIs.

 4. We could probably figure out some new configuration option to add to ZeroClipboard that would be a `function` responsible for locating some element other than the clipped element (e.g. the clipped element's parent "card" element in your JSFiddle's case) and then applying a class to it.
     ```js
ZeroClipboard.config({
  ancestorHoverClass: "zeroclipboard-is-hover-ancestor",
  ancestorFinder: function(clippedEl) {
    return $(clippedEl).parents(".card").get(0);
  }
});
```
     However, as of now, that idea is very vague and wishy-washy in my head, so I'm disinclined to go that route unless you can offer a more concrete implementation suggestion.

**Comment by rothanachoun** [comment]
Thanks for your solution :) !

**Comment by JamesMGreene** [comment]
- [ ] Option 1: Up to you.
 - [ ] Option 2: Up to you, though requires the same fixes as #463 
 - [ ] Option 3: Up to you.
 - [ ] Option 4: For now, calling this one a `wontfix`
